rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,506 @@
+        if value == 'openTrove':
+            return TroveOperation.OPENTROVE
+        if value == 'closeTrove':
+            return TroveOperation.CLOSETROVE
+        if value == 'adjustTrove':
+            return TroveOperation.ADJUSTTROVE
+        if value == 'accrueRewards':
+            return TroveOperation.ACCRUEREWARDS
+        if value == 'liquidateInNormalMode':
+            return TroveOperation.LIQUIDATENORMALMODE
+        if value == 'liquidateInRecoveryMode':
+            return TroveOperation.LIQUIDATERECOVERYMODE
+        if value == 'redeemCollateral':
+            return TroveOperation.REEDEM
+        # else
+        raise DeserializationError(f'Encountered unknown TroveOperation value {value}')
+
+    def __str__(self) -> str:
+        if self == TroveOperation.OPENTROVE:
+            return 'Open Trove'
+        if self == TroveOperation.CLOSETROVE:
+            return 'Close Trove'
+        if self == TroveOperation.ADJUSTTROVE:
+            return 'Adjust Trove'
+        if self == TroveOperation.ACCRUEREWARDS:
+            return 'Accrue Rewards'
+        if self == TroveOperation.LIQUIDATENORMALMODE:
+            return 'Liquidation In Normal Mode'
+        if self == TroveOperation.LIQUIDATERECOVERYMODE:
+            return 'Liquidation In Recovery Mode'
+        if self == TroveOperation.REEDEM:
+            return 'Redeem Collateral'
+        # else
+        raise AssertionError(f'Invalid value {self} for TroveOperation')
+
+
+class LiquityStakeEventType(Enum):
+    CREATE = 1
+    INCREASE = 2
+    DECREASE = 3
+    REMOVE = 4
+    WITHDRAW = 5
+
+    @staticmethod
+    def deserialize(value: str) -> 'LiquityStakeEventType':
+        if value == 'stakeCreated':
+            return LiquityStakeEventType.CREATE
+        if value == 'stakeIncreased':
+            return LiquityStakeEventType.INCREASE
+        if value == 'stakeDecreased':
+            return LiquityStakeEventType.DECREASE
+        if value == 'stakeRemoved':
+            return LiquityStakeEventType.REMOVE
+        if value == 'gainsWithdrawn':
+            return LiquityStakeEventType.WITHDRAW
+        # else
+        raise DeserializationError(f'Encountered unknown LiquityStakeEventType value {value}')
+
+    def __str__(self) -> str:
+        if self == LiquityStakeEventType.CREATE:
+            return 'Stake Created'
+        if self == LiquityStakeEventType.INCREASE:
+            return 'Stake Increased'
+        if self == LiquityStakeEventType.DECREASE:
+            return 'Stake Decreased'
+        if self == LiquityStakeEventType.REMOVE:
+            return 'Stake Removed'
+        if self == LiquityStakeEventType.WITHDRAW:
+            return 'Gains Withdrawn'
+        # else
+        raise AssertionError(f'Invalid value {self} for LiquityStakeEventType')
+
+
+@dataclass(frozen=True)
+class LiquityEvent:
+    kind: Literal['stake', 'trove']
+    tx: str
+    address: str
+    timestamp: Timestamp
+
+    def _asdict(self) -> Dict[str, Any]:
```

Comment:
I didn't use namedtuple because I have two other that have this as base and namedtuples can't have subclasses  https://stackoverflow.com/a/50369898/2588566

Reply:
But then why not call it `serialize()`? Why call it `_asdict`?

Reply:
Because in the function we have to transform data for the API's responses I included it in the classes that use `dict` instead of serialize. The reason is that at the beginning I was using namedtuple and then I just went forward with the `dict` requirement. I can change it to serialize

Reply:
That would probably be more obvious, so yeah please do it.